### PR TITLE
doc: release-notes-4.0: Miscellaneous Cleanup

### PR DESCRIPTION
**Pull request by mmahadevan108**
1. Pulled in a PR from Nordic that had merge conflicts. https://github.com/zephyrproject-rtos/zephyr/pull/81448
2. Move deprecated items to the deprecated section.
3. Fixed a portion of a sentence that was deleted.
4. Miscellaneous cleanup

### CODE REVIEW

**Review by kartben — DISMISSED** [review]

**Review by jhedberg — DISMISSED** [review]

**Comment by kartben** [comment]
@mmahadevan108 actually, if you have a chance, can you change [this ref](https://github.com/nxp-upstream/zephyr/blob/5d3caabf5d8935e57f259daa3846fc794abf9619/doc/releases/release-notes-4.0.rst?plain=1#L416) to be:

```
  be found in the Zephyr tree) to the :ref:`Devicetree bindings <devicetree_binding_index>` documentation.
```

**Review by jhedberg — DISMISSED** [review]

**Comment by kartben** [comment]
@mmahadevan108 thank you!

**Review by kartben — DISMISSED** [review]

**Review by jhedberg — APPROVED** [review]

**Review by EmilioCBen — APPROVED** [review]

**Review by thedjnK — APPROVED** [review]

File: doc/releases/release-notes-4.0.rst [hunk 1]
```diff
@@ -465,17 +480,16 @@ Drivers and Sensors
...
 
   * Added support for STM32L4 AES.
-  * Deprecated the TinyCrypt shim driver ``CONFIG_CRYPTO_TINYCRYPT_SHIM``.
 
 * DAC
 
   * DAC API now supports specifying channel path as internal. Support has been added in STM32 drivers.
-  * Updated the NXP counter_mcux_lptmr driver to support multiple instances of the lptmr
-    peripheral.
   * Updated the initialization of clocks for the NXP LPTMR driver
```
Comment (danieldegrasse) [inline doc/releases/release-notes-4.0.rst:493]: These lines (until 496) also need to be moved to the `counter` section- also:
- the NXP software watchdog timer seems to be a watchdog driver, not a counter or DAC driver. So the note about that should probably go into the watchdog section
- the `nxp,s32-sys-timer` is a timer driver, it should be in the "timer" section

Comment (mmahadevan108) [reply doc/releases/release-notes-4.0.rst:493]: Thanks for catching this. PR https://github.com/zephyrproject-rtos/zephyr/pull/81472

**Pull request merged**